src/theory/arith/nl/ext/factoring_check.cpp

```diff
@@ -120,7 +121,13 @@ void FactoringCheck::check(const std::vector<Node>& asserts,
           sum = Rewriter::rewrite(sum);
           Trace("nl-ext-factor")
               << "* Factored sum for " << x << " : " << sum << std::endl;
-          Node kf = getFactorSkolem(sum);
+
+          LazyCDProof* proof = nullptr;
```

Comment:
Any reason why this proof is not just allocated more local within getFactorSkolem?